src/components/playbackManager.ts

```diff
@@ -204,46 +223,92 @@ export class playbackManager {
         loadRequestData.media = mediaInfo;
         loadRequestData.autoplay = true;
 
+        // If we should seek at the start, translate it
+        // to seconds and give it to loadRequestData :)
+        if (mediaInfo.customData.startPositionTicks > 0)
+            loadRequestData.currentTime =
+                mediaInfo.customData.startPositionTicks / 10000000;
+
         load($scope, mediaInfo.customData, item);
         this.playerManager.load(loadRequestData);
 
+        console.log('setting src to ' + streamInfo.url);
         $scope.PlaybackMediaSource = mediaSource;
-
-        console.log('setting src to ' + url);
         $scope.mediaSource = mediaSource;
+        $scope.audioStreamIndex = streamInfo.audioStreamIndex;
+        $scope.subtitleStreamIndex = streamInfo.subtitleStreamIndex;
 
         DocumentManager.setPlayerBackdrop(item);
 
-        reportPlaybackStart($scope, getReportingParams($scope));
-
         // We use false as we do not want to broadcast the new status yet
         // we will broadcast manually when the media has been loaded, this
         // is to be sure the duration has been updated in the media element
         this.playerManager.setMediaInformation(mediaInfo, false);
     }
 
-    stop(continuing = false): Promise<any> {
-        $scope.playNextItem = continuing;
-        stop();
+    /**
+     * stop playback, as requested by the client
+     */
+    static stop(): void {
+        this.playerManager.stop();
+        // onStopped will be called when playback comes to a halt.
+    }
 
-        const reportingParams = getReportingParams($scope);
+    /**
+     * Called when media stops playing.
+     * TODO avoid doing this between tracks in a playlist
+     */
+    static onStopped(): void {
+        if (this.getNextPlaybackItemInfo()) {
+            $scope.playNextItem = true;
+        } else {
+            $scope.playNextItem = false;
```

Comment:
```suggestion
    static onStop(): void {
        if (this.getNextPlaybackItemInfo()) {
            $scope.playNextItem = true;
        } else {
            $scope.playNextItem = false;
```

Reply:
just skipping this for now

Reply:
Fixed [here](https://github.com/hawken93/jellyfin-chromecast/pull/228/commits/3fc8967c3ac66b612033e9b2dd09c01cf5168e6f)